cli/bpmetadata/cmd.go

```diff
@@ -77,8 +77,22 @@ func generate(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	for _, path := range allBpPaths {
-		err = generateMetadataForBpPath(path)
+	for _, modPath := range allBpPaths {
+		// check if module path has readme.md
+		_, err := os.Stat(path.Join(modPath, readmeFileName))
+
+		// throw an error and exit if root module doesn't have a readme.md
+		if !strings.Contains(modPath, modulesPath+"/") && err != nil {
+			return fmt.Errorf("Top-level module does not have a readme. Details: %w\n", err)
+		}
+
+		// log info if a sub-module doesn't have a readme.md and continue
+		if err != nil {
+			Log.Info("Skiping metadata for sub-module", "Path:", modPath)
```

Comment:
So users are aware of why
```suggestion
			Log.Info("Skipping metadata for sub-module identified as an internal module", "Path:", modPath)
```

Reply:
done